Bio/SeqIO/AbiIO.py

```diff
@@ -401,18 +401,33 @@ def AbiIterator(handle, alphabet=None, trim=False):
     # raw data (for advanced end users benefit)
     annot['abif_raw'] = raw
 
-    # use the file name as SeqRecord.name if available
-    try:
-        file_name = basename(handle.name).replace('.ab1', '')
-    except AttributeError:
-        file_name = ""
-    record = SeqRecord(Seq(seq, alphabet),
-                       id=sample_id, name=file_name,
-                       description='',
-                       annotations=annot,
-                       letter_annotations={'phred_quality': qual})
-
-    if not trim:
+    # fsa check
+    if (set(['SpNm1', 'LIMS1', 'CTID1']).issubset(raw)):
```

Comment:
Since `set(['SpNm1', 'LIMS1', 'CTID1']).issubset(raw)` is used twice, I would put the result in a variable instead.

Also, I take it this means that it is not expected for any ab1 and/or fsa files to have just any of these tags?